templates/diskovc/diskovc.py

```diff
@@ -0,0 +1,162 @@
+            raise j.exceptions.Input("total and read/write of bytes_sec_max cannot be set at the same time")
+
+        if data['totalIopsSecMax'] and (data['readIopsSecMax'] or data['writeIopsSecMax']):
+            raise j.exceptions.Input("total and read/write of iops_sec_max cannot be set at the same time")
+
+
+    def update_data(self, data):
+        # merge new data
+        self.data.update(data)
+        
+        self.save()
+
+    @property
+    def sshkey(self):
+        """ Get a path and keyname of the sshkey service """
+
+        sshkeys = self.api.services.find(template_uid=self.SSH_TEMPLATE)
+        if len(sshkeys) != 1:
+            raise RuntimeError('found %s ssh services, requires exactly 1' % len(sshkeys))
+
+        # Get key name and path
+        path = sshkeys[0].data['path']
+        key = path.split('/')[-1]
+  
+        return key 
+
+    @property
+    def ovc(self):
+        """
+        An ovc connection instance
+        """
+        if self._ovc is not None:
+            return self._ovc
+
+        self._ovc = j.clients.openvcloud.get(self.data['openvcloud'])
+        return self._ovc        
+
+    @property
+    def account(self):
+        """
+        Return an account
+        """
+        if self._account is not None:
+            return self._account
+
+        # Get object for an OVC service, make sure exactly one is running
+        accounts = self.api.services.find(template_uid=self.ACCOUNT_TEMPLATE, name=self.data.get('account', None))
+        if len(accounts) != 1:
+            raise RuntimeError('found %s openvcloud connections, requires exactly 1' % len(accounts))
+        account = accounts[0].name  
+        self._account = self.ovc.account_get(account, create=True)
+
+        return self._account
+
+    def create(self):
+        data = self.data
+        ovc = self.ovc
+        account = self.account
+
+        guid = [location['gid'] for location in ovc.locations if location['name']==data['location']]
+        if not guid:
+            raise RuntimeError('location "%s" not found'%data['location'])
+
+        data['diskId'] = account.disk_create(
+                            name=data['devicename'],
+                            gid=guid,
+                            description=data['description'],
+                            size=data['size'],
+                            type=data['type'],
+                        )
+        import ipdb; ipdb.set_trace()
+        self.save()        
+
+
+    def delete(self):
+        import ipdb; ipdb.set_trace()
+        data = self.data
+        account = self.account
+
+        if data['diskId'] not in [disk['id'] for disk in account.disks]:
+            raise j.exceptions.AYSNotFound(
+                "Data Disk was not found. cannot continue init of %s" % self.template_name
+                )
+        
+        account.disk_delete(data['diskId'])
+
+    
+    def limit_io(self):
```

Comment:
This should not be a separate method, but just called after the disk creation or when the limits changes